### PR TITLE
feat(expr): access `jsonb` object field and array element

### PR DESCRIPTION
**Pull request by xiangjinwu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Adds the following expressions:
* `jsonb_object_field(jsonb, varchar) -> jsonb`
* `jsonb_array_element(jsonb, int) -> jsonb`
* `jsonb_object_field_text(jsonb, varchar) -> varchar`
* `jsonb_array_element_text(jsonb, int) -> varchar`
* `jsonb_typeof(jsonb) -> varchar`
* `jsonb_array_length(jsonb) -> int`

The first two are actually operator `->` in PostgreSQL, and the two in the middle are operator `->>` in PostgreSQL. But our parser does not support parsing this syntax yet.

The optimization of constant rhs will be added in a followup.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added an item to track fuzzing tests.
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

- SQL commands, functions, and operators

### Release note

Adds the following expressions:
* `jsonb_object_field(jsonb, varchar) -> jsonb`
* `jsonb_array_element(jsonb, int) -> jsonb`
* `jsonb_object_field_text(jsonb, varchar) -> varchar`
* `jsonb_array_element_text(jsonb, int) -> varchar`
* `jsonb_typeof(jsonb) -> varchar`
* `jsonb_array_length(jsonb) -> int`

### CODE REVIEW

File: src/expr/src/vector_op/jsonb_info.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        .write_str(v.type_name())
+        .map_err(|e| ExprError::Internal(e.into()))
+}
+
+#[inline(always)]
+pub fn jsonb_array_length(v: JsonbRef<'_>) -> Result<i32> {
+    v.array_len()
+        .map(|n| n as i32)
+        .map_err(|e| ExprError::InvalidParam {
+            name: "",
```
Comment (BugenZhao) [inline src/expr/src/vector_op/jsonb_info.rs:33]: Should we fill this?

File: src/expr/src/expr/expr_jsonb_access.rs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    } else {
+        p as usize
+    };
+    v.access_array_element(idx)
+}
+
+trait AccessOutput: ArrayBuilder {
+    fn return_type() -> DataType;
+    fn output(&mut self, v: JsonbRef<'_>) -> crate::Result<()>;
+    fn output_datum(v: JsonbRef<'_>) -> Datum;
```
Comment (BugenZhao) [inline src/expr/src/expr/expr_jsonb_access.rs:168]: What about naming it `to_datum` as its behavior differs from `output` and `output_nullable`? 🤔

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: src/expr/src/vector_op/jsonb_info.rs (hunk 1 above)
Comment (xiangjinwu) [reply src/expr/src/vector_op/jsonb_info.rs:33]: It is a unary function and the message is readable without a parameter name:
```
dev=> select jsonb_array_length('true');
ERROR:  QueryError: Expr error: Invalid parameter : cannot get array length of a jsonb boolean
```
tbh I just picked an error variant that has the closest meaning and accepts free-form error string 😂 In PostgreSQL it is simply:
```
test=# select jsonb_array_length('true');
ERROR:  cannot get array length of a scalar
```

Comment (BugenZhao) [reply src/expr/src/vector_op/jsonb_info.rs:33]: There's an additional space after "parameter"! 😂

**Pull request merged**